js/main.js

```diff
@@ -50,18 +57,84 @@ function isWeaponFilteredOut(weaponName, isOffhand) {
         case 'Small Shield':
         case 'Medium Shield':
         case 'Greatshield':
+        case 'Thrusting Shield':
             return FILTER_NO_MAINHAND_SHIELD.checked;
         case 'Glintstone Staff':
             return FILTER_NO_MAINHAND_STAFF.checked;
         case 'Sacred Seal':
             return FILTER_NO_MAINHAND_SEAL.checked;
+        case 'Light Bow': 
+        case 'Bow':
+        case 'Greatbow':
         case 'Crossbow':
-            return FILTER_NO_MAINHAND_CROSSBOW.checked;
+            return FILTER_NO_MAINHAND_BOW.checked;
+        case 'Torch':
+            return FILTER_NO_MAINHAND_TORCH.checked;
+        case 'Perfume Bottle':
+            return FILTER_NO_MAINHAND_PERFUME_BOTTLE.checked;
+    }
+    return false;
+}
+function enableCheckboxes(checkboxId, isEnabled){
+    const checkbox = document.getElementById(checkboxId)
+    if(!checkbox){
+        throw new Error("Failed to find checkbox with id" + checkboxID);
+    }
+    checkbox.disabled = !isEnabled;
+}
+
+function toggleDLCCheckboxes(){
+    toggleDLCWeaponCheckbox(INCLUDE_DLC.checked);
+    toggleDLCFilter(INCLUDE_DLC.checked);
+}
+
+function toggleDLCFilter(isEnabled){
+    NO_BASE.disabled = !isEnabled;
+    FILTER_NO_MAINHAND_PERFUME_BOTTLE.disabled = !isEnabled;
+    if(!isEnabled){
+        NO_BASE.checked = false;
+        FILTER_NO_MAINHAND_PERFUME_BOTTLE.checked = false;
+    }
+}
+
+function toggleDLCWeaponCheckbox(isEnabled){
+    const checkboxes = [
+        "checkbox-backhand-blade",
+        "checkbox-beast-claw",
+        "checkbox-great-katana",
+        "checkbox-hand-to-hand-art",
+        "checkbox-light-greatsword",
+        "checkbox-perfume-bottle",
+        "checkbox-throwing-blade",
+        "checkbox-thrusting-shield",
+    ];
+    checkboxes.forEach(checkboxId => enableCheckboxes(checkboxId, isEnabled));
+}
+
+//checks both ashes and weapons, loaded in dlc-exclusives
+function isDLCExclusive(name){
+    if(INCLUDE_DLC.checked){
+        return false;
+    }
+    return DLC_EXCLUSIVES[name];
+}
+
+function onlyDLCExclusives(name){
+    if(NO_BASE.checked){
+        return !DLC_EXCLUSIVES[name]; 
     }
     return false;
 }
 
 function isWeaponUsable(weaponName, isOffhand) {
+    // include dlc weapons
+    if(isDLCExclusive(weaponName)) {
+        return false;
+    }
+    // excludes non dlc weapons
+    if(onlyDLCExclusives(weaponName)) {
+        return false
+    }
```

Comment:
These code comments are redundant; the function names should be self-documenting.

When that is done, this should also be simplified to
```suggestion
    if (isDLCExlusive(weaponName) || onlyDLCExclusives(weaponName)) {
        return false;
    }
```

Reply:
The comments are left over, when function names were not set in stone. Adjusted that part of the code.